src/state/player.ts

```diff
@@ -14,169 +15,252 @@ export interface Timestamp {
+
+				// Load incomming show into media session
+				// Side note: the mediaSession API is neat
+				// https://developer.mozilla.org/en-US/docs/Web/API/MediaSession
+				load_media_session(incoming_show);
+
+				// Starts timer that save listening position.
+				// We're listening to this because the event from audio is firing too often and set to 0 when a new episode is loaded. This works better.
+				start_save_position_interval(incoming_show.number);
+
+				save_state_to_indexed_db();
+
+				// This opens the UI Player drawer
+				player_window_status.set('ACTIVE');
+
+				// Finally Start Playing
+				this.play();
+			} catch (error) {
+				update((state) => ({ ...state, status: 'INITIAL' }));
+			}
+		},
 
-		if (show) {
+		play() {
+			// On play, update the state writable and play audio
 			update((state) => {
-				state.current_show = show;
+				if (state.audio) {
+					state.audio.play();
+				}
+				state.status = 'PLAYING';
+				if (state?.current_show?.number) start_save_position_interval(state.current_show.number);
+				return state;
+			});
+		},
 
-				state.audio?.play().then(() => {
-					// Wait for the audio to be ready to play before setting the new timestamp
-					if (state.audio) state.audio.currentTime = time_stamp;
-				});
+		pause() {
+			stop_save_position_interval();
+			// On pause, update the state writable and play audio
+			update((state) => {
+				if (state.audio) {
+					state.audio.pause();
+				}
+				state.status = 'PAUSED';
 				return state;
 			});
-			player_window_status.set('ACTIVE');
-		}
-	}
+		},
 
-	function close() {
-		player_window_status.set('HIDDEN');
-		update((state) => {
-			if (state.audio) {
-				state.audio.pause();
-				state.audio.src = '';
-				state.audio.crossOrigin = null;
-				state.audio.currentTime = 0;
-			}
+		reset() {
+			// Resetting the player state.
+			// First stop the local storage save interval
+			stop_save_position_interval();
 
-			state.current_show = null;
-			state.playing = false;
-			state.currentTime = 0;
-			state.status = 'INITIAL';
+			//  Reset the player state and pause audio
+			// Set currentTime to 0 (probably doesn't need to happen)
+			update((state) => {
+				if (state.audio) {
+					state.audio.pause();
+					state.audio.currentTime = 0;
+				}
+				return { ...initial_state, audio: state.audio, media_controller: state.media_controller };
+			});
+		},
 
-			return state;
-		});
-	}
+		// Jumps the time in the playing show
+		update_time(time: number) {
+			update((state) => {
+				if (state.audio) {
+					state.audio.currentTime = time;
+				}
+				return state;
+			});
+		},
+
+		close() {
+			stop_save_position_interval();
+			update((state) => {
+				if (state.audio) {
+					if (state.current_show) {
+						localStorage.setItem(
+							`last_played_position_${state.current_show.number}`,
+							state.audio.currentTime.toString()
+						);
+					}
+					state.audio.pause();
+					state.audio.src = '';
```

Comment:
You should also add `state.audio.removeAttribute('src')` = setting to a blank string will set the src to the page url, which can cause it to try request the current page as audio in the future.

Reply:
Update in next